addons/mail/models/res_users_settings_volumes.py

```diff
@@ -31,14 +31,11 @@ def _discuss_users_settings_volume_format(self):
         return [{
             'id': volume_setting.id,
             'volume': volume_setting.volume,
-            'guest_id': {
-                'id': volume_setting.guest_id.id,
-                'name': volume_setting.guest_id.name,
-            } if volume_setting.guest_id else False,
-            'partner_id': {
-                'id': volume_setting.partner_id.id,
-                'name': volume_setting.partner_id.name,
-            } if volume_setting.partner_id else False,
+            'persona': {
+                'id': volume_setting.partner_id.id if volume_setting.partner_id else volume_setting.guest_id.id,
+                'name': volume_setting.partner_id.name if volume_setting.partner_id else volume_setting.guest_id.name,
+                'type': "partner" if volume_setting.partner_id else "guest"
+            } if volume_setting.guest_id or volume_setting.partner_id else False,
```

Comment:
last condition is not necessary, there is constraint to have either guest or partner on volume